test/model/src/core/kv_table.cpp

```diff
@@ -227,6 +227,59 @@ kv_table::remove(kv_transaction_ptr txn, const data_value &key)
     }
 }
 
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
+ */
+int
+kv_table::truncate(const data_value &start, const data_value &stop, timestamp_t timestamp)
+{
+    std::lock_guard lock_guard(_lock);
+    if (start != model::NONE && stop != model::NONE && start > stop)
+        throw model_exception("The start and the stop key are not in the right order");
+
+    auto start_iter = start == model::NONE ? _data.begin() : _data.lower_bound(start);
+    auto stop_iter = stop == model::NONE ? _data.end() : _data.upper_bound(stop);
+
+    try {
+        for (auto i = start_iter; i != stop_iter; i++)
+            i->second.add_update(
+              std::move(kv_update(NONE, fix_timestamp(timestamp))), false, false);
+    } catch (wiredtiger_exception &e) {
+        return e.error();
```

Comment:
Should we `throw` an error instead of returning a value? If we do this, we can make this function `void`.

Reply:
I ended up standardizing the code to return `int` in most cases. As these operations are supposed to behave like their WT counterparts, I ended up setting up the API so that the return codes are directly comparable. (Even if it feels backwards to turn an exception into an error code, and then later throw an exception on error in some but not all cases.)

That being said, it is a good question how much this design decision makes sense. Maybe it would be worth switching everything to `void` as it this point I'm not going all the way to make the API fully WT-like. I guess at this point I'm not far from using these functions the way they have been intended for all along ([WT-11620](https://jira.mongodb.org/browse/WT-11620) and [WT-11621](https://jira.mongodb.org/browse/WT-11621)), at which point I can see what makes more sense.